subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -183,25 +195,33 @@ public void afterStart(GradleInternal gradle) {
                             FileResolver fileResolver = new BaseDirFileResolver(startParameter.getCurrentDir(), () -> {
                                 throw new UnsupportedOperationException();
                             });
-                            List<File> changedPathsSinceLastBuild = getChangedPathsSinceLastBuild(fileResolver, systemPropertiesArgs);
-                            for (File changedPathSinceLastBuild : changedPathsSinceLastBuild) {
-                                LOGGER.warn("Marking as changed since last build: {}", changedPathSinceLastBuild);
+                            if (getSystemProperty(VFS_DROP_PROPERTY, systemPropertiesArgs) != null) {
+                                virtualFileSystem.invalidateAll();
+                            } else {
+                                List<File> changedPathsSinceLastBuild = getChangedPathsSinceLastBuild(fileResolver, systemPropertiesArgs);
```

Comment:
I think we know should remove the ability to pass file changes via a system property.

Reply:
It's still used by the profiler.

Reply:
It is, but it shouldn't, right? We should wait the necessary time between invocations.

Reply:
Not really. It's already slow enough to run a benchmark. It would quadruple the necessary time without additional benefits. The benchmark's goal is not to test the functionality of the watcher.

Once we have swift native watchers I think we should reëvaluate this.

Reply:
Let's do this in a follow up PR.